### PR TITLE
Added dim to portal values

### PR DESCRIPTION
**Pull request by KiloJoel**
This adds: Dimension is saved for portals, so it will teleport the player back to the correct dim when exiting a pocket plane.

Portals created before this change will continue to function, and will still send the player to the overworld, since no dim value is saved.

### CODE REVIEW

**Comment by KiloJoel** [comment]
Fixes an issue in https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/7338

File: src/main/java/com/kentington/thaumichorizons/common/blocks/BlockPortalTH.java [hunk 1]
```diff
@@ -95,29 +96,37 @@ public void onEntityCollidedWithBlock(final World world, final int x, final int
...
                         targetZ = PocketPlaneData.planes.get(planeNum).portalC[2];
+                        if (PocketPlaneData.planes.get(planeNum).portalC.length > 3)
+                            targetDim = PocketPlaneData.planes.get(planeNum).portalC[3];
                         break;
                     }
                     case 3: {
                         targetX = PocketPlaneData.planes.get(planeNum).portalD[0];
                         targetY = PocketPlaneData.planes.get(planeNum).portalD[1] - 2;
                         targetZ = PocketPlaneData.planes.get(planeNum).portalD[2];
+                        if (PocketPlaneData.planes.get(planeNum).portalD.length == 4)
```
Comment (repo-alt) [inline src/main/java/com/kentington/thaumichorizons/common/blocks/BlockPortalTH.java:123]: Not that it really matters but why it is ==4 here and <3 in other places? Also if you got to it, why not compact that weird switch?

**Review by repo-alt — APPROVED** [review]

File: src/main/java/com/kentington/thaumichorizons/common/blocks/BlockPortalTH.java (hunk 1 above)
Comment (KiloJoel) [reply src/main/java/com/kentington/thaumichorizons/common/blocks/BlockPortalTH.java:123]: Ah whoops yeah, missed that, thanks.

And yeah I guess I'll compact it, I'll make another commit in a bit.

**Comment by KiloJoel** [comment]
Not finished yet

**Comment by Dream-Master** [comment]
you know i have fast fingers klick on merge multiple times :)

**Comment by KiloJoel** [comment]
Finished now

**Comment by Dream-Master** [comment]
@repo-alt ok now ?

**Review by repo-alt — APPROVED** [review]

**Pull request merged**